### PR TITLE
Alerting: Allow filtering of contact points by name

### PR DESCRIPTION
**Pull request by alexweav**
**What this PR does / why we need it**:

This PR allows filtering of contact points by name through a query parameter, e.g. `GET /api/v1/provisioning/contact-points?name=blah`

**Which issue(s) this PR fixes**:

Supports https://github.com/grafana/grafana/issues/36153

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by grobinson-grafana** [comment]
If we don't intend to support more than one name i.e. `GET /api/v1/provisioning/contact-points?name=foo&name=bar` shouldn't we just do `GET /api/v1/provisioning/contact-points/:name`?

File: pkg/services/ngalert/provisioning/contactpoints.go [hunk 1]
```diff
@@ -34,17 +34,26 @@ func NewContactPointService(store AMConfigStore, encryptionService secrets.Servi
 	}
 }
 
-func (ecp *ContactPointService) GetContactPoints(ctx context.Context, orgID int64) ([]apimodels.EmbeddedContactPoint, error) {
-	revision, err := getLastConfiguration(ctx, orgID, ecp.amStore)
+type ContactPointQuery struct {
+	Name  string
```
Comment (JohnnyQQQQ) [inline pkg/services/ngalert/provisioning/contactpoints.go:38]: We should document that this is optional.

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM

**Comment by JohnnyQQQQ** [comment]
@grobinson-grafana this is already used for `uid`

File: pkg/services/ngalert/provisioning/contactpoints.go (hunk 1 above)
Comment (alexweav) [reply pkg/services/ngalert/provisioning/contactpoints.go:38]: Done

I had a thought along these lines - perhaps we should make orgID a separate parameter to GetContactPoints and not put in the query? It's definitely required and it'd be more type safe to force the caller to deal with the parameter. Go doesn't force you to fill in any struct fields, so someone could miss it.

**Comment by alexweav** [comment]
@grobinson-grafana We can always extend this to support multiple params in the future without breaking compatibility, if we want.

If we put it in the path, we can no longer do this.

File: pkg/services/ngalert/provisioning/contactpoints.go (hunk 1 above)
Comment (alexweav) [reply pkg/services/ngalert/provisioning/contactpoints.go:38]: I'm a little torn on this. I like the simplicity of `GetContactPoints(context, query)` too, it's rather clean.

Comment (alexweav) [reply pkg/services/ngalert/provisioning/contactpoints.go:39]: Went with just the documentation for now for the sake of simplicity. We can always extend this.

**Pull request merged**